holoviews/core/util.py

```diff
@@ -24,6 +26,57 @@
 except ImportError:
     dd = None
 
+
+
+
+class HashGenerator(json.JSONEncoder):
+    """
+    Extends JSONEncoder to generate a hashable string for as many types
+    of key as possible. These keys are supplied by widgets and Stream
+    parameters and may be nested. If an unrecognized object type is
```

Comment:
And I'm confused about whether it's truly a HashGenerator -- the docstring says it will generate a hashable string, not a hash, and the code indicates that sometimes it's one, sometimes it's the other.  Which is it?